navigation/ios/FullScreenNotAnimatedPresenter.swift

```diff
@@ -88,14 +88,12 @@ struct FullScreenNotAnimatedPresenter<Content: View>: UIViewRepresentable {
     }
 }
 
-extension UIView {
-    var owningController: UIViewController? {
-        if let responder = next as? UIViewController {
-            return responder
-        } else if let responder = next as? UIView {
-            return responder.owningController
+extension UIResponder {
+    var nextViewController: UIViewController? {
+        if let viewControllerResponder = self as? UIViewController {
+            viewControllerResponder
         } else {
-            return nil
+            next?.nextViewController
         }
     }
```

Comment:
You could do that with a loop instead of using recursion.

Reply:
Doesn't swift apply tail call optimization automatically?

Reply:
We checked it does with this code?

Reply:
Not sure, hence my question (kinda noob in swift), I wonder if that is automatic or not!
With my kotlin background I would've used tailrec but I'm wondering if we have to pay attention to this or the compiler does his things.

Reply:
As with all compiler optimizations there are limits. 10 years ago there was a limit when ARC was involved (as it probably is in this code since it return objects). No clue if it was removed.